RedBookPlayer/PlayableDisc.cs

```diff
@@ -0,0 +1,662 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Aaru.CommonTypes.Enums;
+using Aaru.CommonTypes.Structs;
+using Aaru.DiscImages;
+using Aaru.Helpers;
+using static Aaru.Decoders.CD.FullTOC;
+
+namespace RedBookPlayer
+{
+    public class PlayableDisc
+    {
+        #region Public Fields
+
+        /// <summary>
+        /// Indicate if the disc is ready to be used
+        /// </summary>
+        public bool Initialized { get; private set; } = false;
+
+        /// <summary>
+        /// Current track number
+        /// </summary>
+        public int CurrentTrackNumber
+        {
+            get => _currentTrackNumber;
+            set
+            {
+                // Unset image means we can't do anything
+                if(_image == null)
+                    return;
+
+                // Check if we're incrementing or decrementing the track
+                bool increment = value >= _currentTrackNumber;
+
+                // Ensure that the value is valid, wrapping around if necessary
+                if(value >= _image.Tracks.Count)
+                    _currentTrackNumber = 0;
+                else if(value < 0)
+                    _currentTrackNumber = _image.Tracks.Count - 1;
+                else
+                    _currentTrackNumber = value;
+
+                // Cache the current track for easy access
+                Track track = _image.Tracks[CurrentTrackNumber];
+
+                // Set new track-specific data
+                byte[] flagsData = _image.ReadSectorTag(track.TrackSequence, SectorTagType.CdTrackFlags);
+                ApplyDeEmphasis = ((CdFlags)flagsData[0]).HasFlag(CdFlags.PreEmphasis);
+
+                try
+                {
+                    byte[] subchannel = _image.ReadSectorTag(track.TrackStartSector, SectorTagType.CdSectorSubchannel);
```

Comment:
The returned subchannel can not be the one from the sector you want.

Reply:
In case like that, how do you determine where it came from? Or is that something I may have to accept blindly?

Reply:
the subchannel tells you where it comes from, it has a "current position" in Q (partial when it encodes MCN or ISRC)

Reply:
Ah, I think I have it:
`byte[] subchannel = _image.ReadSectorTag(track.TrackStartSector, track.TrackSequence, SectorTagType.CdSectorSubchannel);`. Is my understanding of that overload correct for your comment here?

Reply:
I did not have it. My understanding was incorrect. I'm looking into the subchannel format.